src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -764,6 +770,34 @@ namespace levin
     );
   }
 
+  void notify::on_connection_new(const boost::uuids::uuid &id, bool is_income)
+  {
+    if (!zone_)
+      return;
+
+    const auto zone = zone_;
```

Comment:
We're now aliasing the purpose of the `connection_context` concept in another file, and technically the raciness of the original code was never fixed since the context copies everything including UUID objects, etc.

Just mark the copy+move constructors+operators as deleted, and add a named function that copies the useful fields for the async callback. Ideally the fields copied would be the bare minimum needed (the UUID probably doesn't actually need to be touched, etc).

Reply:
Or let me re-state my argument -

The problematic line is the single (implicit) copy-assignment call. The copy+move constructor+assignment should (almost certainly in hindsight) be deleted from the context object since it is read+write by multiple threads. Or a thread-safe explicit version should be created. Its too easy to add a field and be unaware that there's an unsynchronized copy that occurs elsewhere due to the implicit function.

Reply:
I went back through the code again, and this suggestion is a pain but doable. The duplicate `context` class and tracking logic is unfortunate, but might be the lesser of evils at the moment. Ideally those values are in the "real" context, with copying/moving disabled on that class.

I'll investigate if that call sequence with the copies can be re-thought. The context object is huge to be copied, when only like 5 trivial fields max are needed.